File: src/main/java/yaasl/server/model/RememberMeToken.java
package yaasl.server.model;

import org.springframework.security.web.authentication.rememberme.PersistentRememberMeToken;

import javax.persistence.*;
import java.util.Date;

import static javax.persistence.FetchType.LAZY;
import static javax.persistence.GenerationType.AUTO;

@Entity
public class RememberMeToken  {

    private Long id;
    private String username;
    private String series;
    private String tokenValue;
    private Date date;
    private User user;

    public RememberMeToken() {
    }

    public RememberMeToken(PersistentRememberMeToken rememberMeToken, User user) {
        this.username = rememberMeToken.getUsername();
        this.date = rememberMeToken.getDate();
        this.series = rememberMeToken.getSeries();
        this.tokenValue = rememberMeToken.getTokenValue();
        this.user = user;
    }

    @Id
    @GeneratedValue(strategy=AUTO)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getSeries() {
        return series;
    }

    public void setSeries(String series) {
        this.series = series;
    }

    public String getTokenValue() {
        return tokenValue;
    }

    public void setTokenValue(String tokenValue) {
        this.tokenValue = tokenValue;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @ManyToOne(fetch = LAZY)
    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

}

File: src/main/kotlin/yaasl/server/model/RememberMeToken.kt
package yaasl.server.model

import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.AUTO
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.FetchType.LAZY

@Entity
class RememberMeToken(@get:Id
                      @get:GeneratedValue(strategy = AUTO)
                      var id: Long? = null,
                      var username: String? = null,
                      var series: String? = null,
                      var tokenValue: String? = null,
                      var date: Date? = null,
                      @get:ManyToOne(fetch = LAZY)
                      var user: User? = null)
